src/custom_gates.jl

```diff
@@ -18,6 +43,132 @@ function matrix_rep(g::DoubleExcitation)
     return SMatrix{16,16,ComplexF64}(mat)
 end
 
+"""
+    DoubleExcitationPlus(ϕ)
+
+Generate the matrix representation of the [DoubleExcitationPlus](https://docs.pennylane.ai/en/stable/code/api/pennylane.DoubleExcitationPlus.html) gate.
+
+This gate performs an SO(2) rotation in the subspace {|1100⟩, |0011⟩} with a phase-shift on other states:
+
+- |0011⟩ ⟼ cos(ϕ/2)|0011⟩ - sin(ϕ/2)|1100⟩
+- |1100⟩ ⟼ cos(ϕ/2)|1100⟩ + sin(ϕ/2)|0011⟩
+- |x⟩ ⟼ e^{iϕ/2}|x⟩   for all other basis states |x⟩
+
+# Examples
+
+```jldoctest
+julia> ϕ = 3.56;
+julia> gate_matrix = DoubleExcitationPlus(ϕ);
+julia> m  = matrix_rep(gate_matrix);
+julia> eq1 = m * [1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1];
+julia> eq2 = m * [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1];
+julia> eq1 ==  [exp(im*ϕ/2), 0, 0, cos(ϕ/2), 0, 0, 0, 0, 0, 0, 0, 0, sin(ϕ/2), 0, 0, exp(im*ϕ/2)]
+true
+julia> eq2 ==  [exp(im*ϕ/2), 0, 0, -sin(ϕ/2), 0, 0, 0, 0, 0, 0, 0, 0, cos(ϕ/2), 0, 0, exp(im*ϕ/2)]
+true
+```
+
+"""
+struct DoubleExcitationPlus <: AngledGate{1}
+    angle::NTuple{1,Union{Float64,FreeParameter}}
+    DoubleExcitationPlus(angle::T) where {T<:NTuple{1,Union{Float64,FreeParameter}}} =
+        new(angle)
+end
+Braket.chars(::Type{DoubleExcitationPlus}) = "G2+(ang)"
+Braket.qubit_count(::Type{DoubleExcitationPlus}) = 4
+Base.inv(g::DoubleExcitationPlus) = DoubleExcitationPlus(-g.angle[1])
+Base.:^(g::DoubleExcitationPlus, power::Integer) = power == -1 ? inv(g) : (power == 0 ? DoubleExcitationPlus((0.0,)) : (power < 0 ? inv(g^(-power)) : DoubleExcitationPlus((g.angle[1] * power,))))
```

Comment:
if `power == 0`, why not return `Identity()`?

Reply:
Also, better to use `iszero` for this check

Reply:
Added `iszero `check. Identity() was not present in` BraketSimulator` so used `DoubleExcitationPlus((0.0,))`

Reply:
It's `Braket.I()`, not `Identity()`, which is my mistake and why you couldn't find it :p

Reply:
It's okay, I have changed to `Braket.I()` instead of ` DoubleExcitationPlus((0.0,))` and likewise.